src/checkers/inference/solver/frontend/Lattice.java

```diff
@@ -0,0 +1,96 @@
+package checkers.inference.solver.frontend;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.lang.model.element.AnnotationMirror;
+
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.AnnotationUtils;
+
+/**
+ * Lattice class contains necessary information about qualifier hierarchy for
+ * constraint constraint solving.
+ * 
+ * @author jianchu
+ *
+ */
+public class Lattice {
+
+    private QualifierHierarchy qualHierarchy;
+    public Map<AnnotationMirror, Collection<AnnotationMirror>> subType = AnnotationUtils .createAnnotationMap();
```

Comment:
Um.... I see why these fields are public but not final, and we have `configure()` separate from `constructor`. This is because we want customized initializations for `Lattice` and `TwoTypeLattice`, and we initialize multiple fields in a single method (e.g. `calculateSubSuperTypes()` initialize `superType`, `subType`, `typeToInt`, and `intToType`..).

In this case, can we introduce Builder Pattern to solve this problem? As I think make those fields final and immutable are important, as they are frequently accessed by others in the Solver framework, so it would be better to achieve type safety on the compiler level rather than trust developers will not make mistakes (I would chose human developers as the last one to trust...).

E.g.  we can have a LatticeBuilder for building `Lattice` and `TwoTypeLattice`:
```java
public class LatticeBuilder {
//All fields need to be initialized in Lattice declared here
private Map<> superType;
....;

public Lattice buildLattice(QualifierHierarchy qualHierarchy) {
  //move `Lattice#configure()` logic here

  //Pass all initialized feilds to the Lattice constructor, and make them final and immutable in the constructor.
  return new Lattice(superType, subType, ....);
 }

 public TwoTypeLattice buildTwoTypeLattice(AM top, AM btm) {
 //move `TwoTypeLattice#configure()` logic here

 //Pass all initialized feilds to the Lattice constructor, and make them final and immutable in the constructor.
 return new TwoTypeLattice(superType, ...);
}


//general lattice
private void calculateSubSuperTypes() {...}

//two type lattice
private void calculateSubSuperTypesForTwoTypeLattice() {...}

//other calculating functions...
}

The reason of proposing Builder pattern is I expect `Lattice` would not have too much sub-classes (currently only `Lattice` and `TwoTypeLattice`). Otherwise `LatticeBuilder` will expend to a god  class that has too many functions.